### ISSUE TITLE
Remove Boost dependency

### ISSUE THREAD

**Issue by Narann** [mupen64plus/mupen64plus-video-glide64mk2#46]
Some [downstream devs](https://github.com/mupen64plus/mupen64plus-core/issues/60#issuecomment-71331446) would like to remove the boost dependency from GlideHQ. This ticket has been created to follow this.

**Comment by littleguy77** [comment]
@Narann @twinaphex @richard42 

So I started looking at this last night.  You probably noticed, boost is only being used for its filesystem library, for just a handful of features:
 - concatenate paths `boost::filesystem::wpath operator /=`
 - determine if a file exists `boost::filesystem::exists(...)`
 - create directory `boost::filesystem::create_directory(...)`
 - iterate over files in a directory `boost::filesystem::directory_iterator`

Easy enough to replace, just pluck the operating system abstraction layer files (`osal_files.h`,  `osal_files_unix.c`, `osal_files_win32.c`) from some of the other mupen64plus plugins.

Except for one complexity: GlideHQ supports unicode filenames.  This is why boost was used in the first place:
```
  /* NOTE: I could use the boost::wdirectory_iterator and boost::wpath
   * to resolve UNICODE file names and paths. But then, _wfopen() is
   * required to get the file descriptor for MS Windows to pass into
   * libpng, which is incompatible with Win9x. Win9x's fopen() cannot
   * handle UNICODE names. UNICODE capable boost::filesystem is available
   * with Boost1.34.1 built with VC8.0 (bjam --toolset=msvc-8.0 stage).
   *
   * RULE OF THUMB: NEVER save texture packs in NON-ASCII names!!
   */
```

***Do we really need to support this feature?***

Do we get user complaints about [mupen64plus-video-rice's lack of unicode support](
https://github.com/mupen64plus/mupen64plus-video-rice/blob/eb4475ec4bbcdc313ae8f62048a822215142ef80/src/TextureFilters.cpp#L1295)?

**Comment by Narann** [comment]
Good catch! I think this has been wrote before the mupen64plus port. I don't know if mupen64plus support unicode. I think @richard42 has a better advise about this but as you point, rice doesn't seems to have unicode support.

**Comment by richard42** [comment]
Personally I don't like Boost, so I'm happy to see it removed from GlideHQ.  And I think all the other filesystem code in Mupen64Plus is not unicode-compatible, so it's not much of a loss to lose this 'feature' here too.

**Comment by littleguy77** [comment]
Thanks for the feedback.  I'll remove the boost dependency and unicode support and try to get a PR out soon.

**Comment by mrfixit2001** [comment]
> Thanks for the feedback. I'll remove the boost dependency and unicode support and try to get a PR out soon.

![giphy](https://user-images.githubusercontent.com/38842252/232524151-45caf6e8-9e6a-4934-8c7b-578b40e08238.gif)

This is the only reason boost is needed on my distro... and one of them needs to go...

**Comment by richard42** [comment]
Just for GlideHQ?  It shouldn't be that hard to remove it.

**Comment by Jj0YzL5nvJ** [comment]
Fixed by #130
